.github/workflows/test.yml

```diff
@@ -97,7 +97,7 @@ jobs:
   # Then, modify the `cross-linux` job to run `test` instead of `build`.
   test-linux:
     name: Unit - x86_64-unknown-linux-gnu
-    runs-on: [linux, test-runner]
+    runs-on: [linux, optimize-8-core]
```

Comment:
Looks like this is still 8; I'm curious what 4 looks like.

Reply:
Ah, nvm, I see you tried it with an earlier commit. I think we could maybe avoid the OOM by restricting the number of concurrent `cargo` jobs.

Reply:
Note that normally `cargo` auto-scales the number of jobs to the number of threads available. What really hits the OOM are the number of link jobs, which requires a special hack to restrict as there is no option currently.